libs/common/result.hpp

```diff
@@ -114,13 +125,45 @@ namespace iroha {
      * result
      */
     template <typename T, typename E, typename Transform>
-    constexpr auto operator|(Result<T, E> r, Transform &&f)
-        -> decltype(f(std::declval<T>())) {
+    constexpr auto operator|(Result<T, E> r, Transform &&f) ->
```

Comment:
As I understand semantic for the `Result` type. `r` has to be `const ref` instead, isn't it?

Reply:
It cannot be, since then we will not be able to apply non-const match, which we might want to do